### PR TITLE
Feature TR-3141 forward port delivery monitoring table merge

### PR DESCRIPTION
**Pull request by wazelin**
# [TR-3141](https://oat-sa.atlassian.net/browse/TR-3141)

## Description
This forward ports:
- [v19.18.5.2](https://github.com/oat-sa/extension-tao-proctoring/releases/tag/v19.18.5.2)
- [v19.18.5.1](https://github.com/oat-sa/extension-tao-proctoring/releases/tag/v19.18.5.1)
- https://github.com/oat-sa/extension-tao-proctoring/pull/896

This is also available as [v19.20.7.2](https://github.com/oat-sa/extension-tao-proctoring/releases/tag/v19.20.7.2).

## How to test

1. Set up `develop` version of the extension
2. Run a few proctored delivery executions and monitoring sessions
3. Run `php tao/scripts/taoUpdate.php` in order to add an extra column to the monitoring table
4. Run `php index.php '\oat\taoProctoring\scripts\tools\MonitoringExtraFieldConfigurationMigration'` to utilize the new column
5. Run `php index.php '\oat\taoProctoring\scripts\tools\MonitoringExtraFieldMigration' -w` in order to migrate the data from `kv_delivery_monitoring` table into the new column of `delivery_monitoring`
6. Compare the values from `kv_delivery_monitoring` with `delivery_monitoring.extra_data` column
7. Run `php index.php '\oat\taoProctoring\scripts\tools\MonitoringExtraFieldMigration' -w -d` in order to delete all the migrated data from `kv_deliery_monitoring`

### CODE REVIEW

**Review by tikhanovichA — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable) 
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by siwane — APPROVED** [review]

**Comment by wazelin** [comment]
Thanks folks!

Will merge right after a verification by the QA engineers.

**Review by yohanlaborda — CHANGES_REQUESTED** [review]

**Comment by wazelin** [comment]
@yohanlaborda, please note all the changes here have been cherry picked from the releases and PRs mentioned in the description.

I will adjust the PR in case you'll find something significant, though.

**Review by siwane — APPROVED** [review]

**Pull request merged**
